### PR TITLE
HAP-1369 - Add support for Pip install in scripted and declarative pipelines

### PR DESCRIPTION
**Pull request by barbelity**
- [x] All [tests](https://ci.appveyor.com/project/jfrog-ecosystem/jenkins-artifactory-plugin) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is created in the [jfrog/jenkins-artifactory-plugin](https://github.com/jfrog/jenkins-artifactory-plugin) repository.
-----

### CODE REVIEW

File: src/main/java/org/jfrog/hudson/pipeline/common/executors/ExtractorJarExecutor.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    BuildInfo buildInfo;
+    Launcher launcher;
+    String javaArgs;
+    FilePath ws;
+    String path;
+    String module;
+    EnvVars env;
+    Run build;
+
+    public ExtractorJarExecutor(BuildInfo buildInfo, Launcher launcher, String javaArgs, FilePath ws, String path, String module, EnvVars env, TaskListener listener, Run build) {
```
Comment (eyalbe4) [inline src/main/java/org/jfrog/hudson/pipeline/common/executors/ExtractorJarExecutor.java:33]: ExtractorJarExecutor --> BuildInfoProcessRunner

Let's also add some description for this class.

File: src/main/java/org/jfrog/hudson/pipeline/common/executors/NpmEnvExtractor.java [hunk 2]
```diff
@@ -29,8 +29,8 @@ public NpmEnvExtractor(Run build, BuildInfo buildInfo, Deployer deployer, Resolv
 
     @Override
     protected void addExtraConfiguration(ArtifactoryClientConfiguration configuration) {
-        configuration.npmHandler.setNpmInstallArgs(args);
-        configuration.npmHandler.setNpmPath(path);
-        configuration.npmHandler.setNpmModule(module);
+        configuration.buildToolHandler.setBuildToolArgs(args);
```
Comment (eyalbe4) [inline src/main/java/org/jfrog/hudson/pipeline/common/executors/NpmEnvExtractor.java:32]: buildToolHandler --> packageManagerHandler

setBuildToolArgs --> setArgs

File: src/main/java/org/jfrog/hudson/pipeline/common/executors/NpmEnvExtractor.java [hunk 3]
```diff
@@ -29,8 +29,8 @@ public NpmEnvExtractor(Run build, BuildInfo buildInfo, Deployer deployer, Resolv
...
+        configuration.buildToolHandler.setBuildToolPath(path);
+        configuration.buildToolHandler.setBuildToolModule(module);
```
Comment (eyalbe4) [inline src/main/java/org/jfrog/hudson/pipeline/common/executors/NpmEnvExtractor.java:34]: setBuildTool... --> setPackageManager...

File: src/main/java/org/jfrog/hudson/pipeline/declarative/steps/pip/PipInstallStep.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        public String getDisplayName() {
+            return "run Artifactory pip install";
+        }
+
+        @Override
+        public boolean isAdvanced() {
+            return true;
+        }
+    }
+
```
Comment (eyalbe4) [inline src/main/java/org/jfrog/hudson/pipeline/declarative/steps/pip/PipInstallStep.java:144]: Redundant new-line.

File: src/test/resources/integration/pipelines/declarative/pip.pipeline [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    stage "Configure Artifactory"
+    rtServer(
+            id: serverId,
+            url: "${env.JENKINS_ARTIFACTORY_URL}",
+            username: "${env.JENKINS_ARTIFACTORY_USERNAME}",
+            password: "${env.JENKINS_ARTIFACTORY_PASSWORD}"
+    )
+
+    stage "Configure npm build"
```
Comment (eyalbe4) [inline src/test/resources/integration/pipelines/declarative/pip.pipeline:20]: npm --> pip

File: src/test/resources/integration/pipelines/scripted/pip.pipeline [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import java.nio.file.Paths
+
+node("TestSlave") {
+    stage "Configure Artifactory"
+    def rtServer = Artifactory.newServer url: "${env.JENKINS_ARTIFACTORY_URL}", username: "${env.JENKINS_ARTIFACTORY_USERNAME}", password: "${env.JENKINS_ARTIFACTORY_PASSWORD}"
+    def buildInfo = Artifactory.newBuildInfo()
+    buildInfo.name = "scripted:pip test"
+    buildInfo.number = "4"
+
+    stage "Configure npm build"
```
Comment (eyalbe4) [inline src/test/resources/integration/pipelines/scripted/pip.pipeline:14]: npm --> pip

File: src/main/java/org/jfrog/hudson/pipeline/common/executors/NpmEnvExtractor.java (hunk 2 above)
Comment (barbelity) [reply src/main/java/org/jfrog/hudson/pipeline/common/executors/NpmEnvExtractor.java:32]: This change will be added in #314 due to a required change in the [build-info project](https://github.com/jfrog/build-info/pull/363).

**Pull request merged**